app/lib/message_tracker.rb

```diff
@@ -0,0 +1,23 @@
+class MessageTracker
+  attr_accessor :client, :message_name
+
+  def initialize(client:, message:)
+    @client = client
+    @message_name = message.name
+  end
+
+  def sent_at
+    return nil unless already_sent?
+
+    DateTime.parse(client.message_tracker[message_name])
+  end
+
+  def already_sent?
+    client.message_tracker[message_name].present?
+  end
+
+  def record(datetime)
```

Comment:
that was the thought -- i had originally thought about limiting so that we only record the value if its a send-only-once message. But, I  thought we might find some value in having the latest sent time 🤔